src/fast-path.js

```diff
@@ -280,17 +282,27 @@ FPp.needsParens = function(assumeExpressionContext) {
       }
 
     case "BinaryExpression":
-      // TODO https://github.com/prettier/prettier/issues/907 this is not complete
+      const isLeftOfAForStatement = node => {
+        let i = 0;
+        while (node) {
+          let parent = this.getParentNode(i++);
+          if (!parent) {
+            return false;
+          }
+          if (parent.type === "ForStatement" && parent.init === node) {
+            return true;
+          }
```

Comment:
Actually, this doesn't work. For some reason VariableDeclaration is considered to be a Statement and therefore

```js
for (let a = (b in c); ; );
```

is no longer parenthesized. I'll leave the optimization off, I don't think it's a big deal going up all the way.